src/main/java/org/shanerx/tradeshop/objects/ShopChest.java

```diff
@@ -157,48 +203,42 @@ public String getName() {
 	}
 
 	public void resetName() {
-		if (chest != null) {
-			BlockState bs = chest.getState();
-			if (bs instanceof Nameable && ((Nameable) bs).getCustomName() != null
-					&& ((Nameable) bs).getCustomName().contains("$ ^Sign:l_")) {
-				((Nameable) bs).setCustomName(((Nameable) bs).getCustomName().split(sectionSeparator)[0]);
-
-				if (isDoubleChest(chest)) {
-					BlockState dblSide = getOtherHalfOfDoubleChest(chest).getState();
-					((Nameable) dblSide).setCustomName(
-							((Nameable) dblSide).getCustomName().split(sectionSeparator)[0]);
+        if (isShopChest(chest)) {
+            Container container = (Container) chest.getState();
+            container.getPersistentDataContainer().remove(plugin.getStorageKey());
+            container.update();
 
-					dblSide.update();
-				}
-
-				bs.update();
+/*
+			if (isDoubleChest(chest)) {
+				Container container2 = (Container)getOtherHalfOfDoubleChest(chest).getState();
+				container2.getPersistentDataContainer().remove(plugin.getStorageKey());
+				container2.update();
 			}
+*/
+
 		}
 	}
 
 	public void setName() {
-		BlockState bs = chest.getState();
-		if (bs instanceof Nameable) {
-			((Nameable) bs).setCustomName(getName());
-
-			if (isDoubleChest(chest)) {
-				BlockState dblSide = getOtherHalfOfDoubleChest(chest).getState();
-				((Nameable) dblSide).setCustomName(getName());
+        setName(chest);
+    }
 
-				dblSide.update();
-			}
+    public void setName(Block toSet) {
+        Container container = (Container) chest.getState();
+        container.getPersistentDataContainer().set(plugin.getStorageKey(), PersistentDataType.STRING, getName());
+        container.update();
 
-			bs.update();
+/*
+		if (isDoubleChest(chest)) {
+			Container container2 = (Container)getOtherHalfOfDoubleChest(chest).getState();
+			container2.getPersistentDataContainer().set(plugin.getStorageKey(), PersistentDataType.STRING, getName());
+			container2.update();
 		}
+*/
```

Comment:
why is the dchest part gone?

Reply:
Wasn't sure if i needed to add the item to both sides of a double chest